src/openrct2/interface/Window_internal.cpp

```diff
@@ -12,28 +12,10 @@ void rct_window::SetLocation(const CoordsXYZ& coords)
 
 void rct_window::ScrollToViewport()
 {
-    // In original checked to make sure x and y were not -1 as well.
-    if (viewport == nullptr || viewport_focus_coordinates.y == -1)
+    if (viewport == nullptr || !focus2.has_value())
         return;
 
-    CoordsXYZ newCoords = {};
-    if (viewport_focus_sprite.type & VIEWPORT_FOCUS_TYPE_SPRITE)
-    {
-        auto* sprite = GetEntity(viewport_focus_sprite.sprite_id);
-        if (sprite == nullptr)
-        {
-            return;
```

Comment:
Does anyone actually clear the focus when the entity vanishes? GetPos introduced log_error I'm not sure if this would continue to spam now if for reasons the entity vanishes.

Reply:
Yeah the smart functions keep the focus entity updated.